### DISCUSSION TITLE
Is there a way to add a row when I check on checkbox

### DISCUSSION THREAD

**Discussion by DanPresa** [KevinVandy/material-react-table#150] [Q&A]
Is there a way to add row object data to an array when I check on each one and add all of them when I check on select all?

And can you give an example please?

I was using accessorFn but. I'd like to know if there is another way.

**Comment by KevinVandy** [comment]
Are you looking for the expanding sub-rows/tree data examples? https://www.material-react-table.com/docs/guides/expanding-sub-rows

**Reply by DanPresa** [reply]
No. I mean, something similar to https://www.material-react-table.com/docs/guides/row-selection#manual-row-selection-without-checkboxes but.
Click on the row an get the whole data pot example `{name: 'Daniel', email: 'dan@mail.com', phone: '12312312}`
And then push that data into an array
```
[
  {
    name: 'Daniel',
    email: dan@mail.com,
    phone: '12312312
  }
]
```

Clink on another row and get the data
```
[
  {
    name: 'Daniel',
    email: dan@mail.com,
    phone: '12312312
  },
  {
    name: 'Kevin',
    email: kev@mail.com,
    phone: '12312312
  }
]
```

**Reply by KevinVandy** [reply]
I mean there are detail panels https://www.material-react-table.com/docs/guides/detail-panel

Otherwise, yeah, it would be easy to hook events to checkboxes based on selection state. Either add an `onChange` to the select checkbox or add a useEffect that watches your `rowSelection` state

**Reply by DanPresa** [reply]
It would be really great to have a feature like that in the future

**Reply by KevinVandy** [reply]
You can already do it. I don't know what you're talking about at this point. The data you pass to the table is up to you

**Reply by DanPresa** [reply]
@KevinVandy 
I'm only trying to do this with your table but. It doesn´t have an options like that.
I did this example with MUI V5

![Capture](https://user-images.githubusercontent.com/39141108/195409122-73905fa1-1e55-425d-bd44-1b20049f6e5d.PNG)

**Reply by KevinVandy** [reply]
Oh so you're looking for an aggregated footer. That's pretty simple and there are some official examples for that already. Have you seen this example yet? https://www.material-react-table.com/docs/examples/aggregation-and-grouping
